### DISCUSSION TITLE
Benefits of compiling via IR

### DISCUSSION THREAD

**Discussion by maikelordaz** [Cyfrin/foundry-full-course-cu#1958] [Q&A]
I still dont get the advantages or disadvantages of compiling via ir. It helps to get rrid of the stack too deep error (although I prefer to refactor and write a better and more readable code when I get this) and it allows to use the custom errors on require statement with the new solidity version 0.8.26. Is there a reason to not use it always? I mean I dont have problems to wait a little bit longer on compiling, as I never faced an issue when this compiling time gets too long, so until now this time have not been a problem for me

**Comment by EngrPips** [comment]
I have not personally compiled a contract using `via IR` myself, so I don't know much about using it or not, but Patrick has a content about it in the `Advanced Foundry Course` on  `Cyfrin Updraft.` Here `https://updraft.cyfrin.io/courses/advanced-foundry/account-abstraction/via-ir?lesson_format=video` is the link to the video maybe check it out and see if it satisfies your curiosity

**Reply by maikelordaz** [reply]
Great, thanks. Actually just yesterday I started seeing the updates in the courses

**Reply by EngrPips** [reply]
You are welcome, Friend. You should catch up with those updates. Cool stuff like `Account Abstraction,` `Airdroping Token Using Merkle Tree,` and others have been added.
